contrib/cirrus/lib.sh.t

```diff
@@ -20,6 +22,11 @@ function check_result {
     fi
 }
 
+function decoration_filter {
+    cat - | tr -d '*>\012'|sed -e 's/^ \+//'
+    exit $?
```

Comment:
I'm still very unhappy about the `pipefail`. I think it may come back to bite us if this test is ever extended. But so be it.

At the very least, since you have to rebase, could you please fix these two lines? First, remove the `cat -`. It is unnecessary (in general, any one-argument use of `cat` should trigger mental alarm bells). Second, remove the `exit`, it is misleading. Thanks.

Reply:
I tried taking out the pipefail...but then the tests fail (when running on my laptop).  I'll add a comment that it's there because it's set in CI.  I made the other changes as well.

Reply:
The `pipefail` isn't necessary if the filtering is done in `check_result()`, as I suggested in my initial post. That would also (bonus) eliminate duplication.